src/main/java/com/softserve/academy/spaced/repetition/security/CustomAuditorAware.java

```diff
@@ -0,0 +1,17 @@
+package com.softserve.academy.spaced.repetition.security;
+
+import com.softserve.academy.spaced.repetition.security.authentification.JwtUser;
+import org.springframework.data.domain.AuditorAware;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+public class CustomAuditorAware implements AuditorAware<Long> {
+    @Override
+    public Long getCurrentAuditor() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication == null) {
+            return null;
+        }
+        return ((JwtUser) authentication.getPrincipal()).getId();
```

Comment:
Can we get `anonymousUser` (String), when call `getPrincipal`?

Reply:
no, because this method is used only with creating object, but action creating have only authorized users,
so other part of security will not resolve creating for anonymousUser